### ISSUE TITLE
[Refactor] U-Net+++ 리팩토링

### ISSUE THREAD

**Issue by yjs616** [boostcampaitech7/level2-cv-semanticsegmentation-cv-02-lv3#64]
## 🤷 작업할 기능 소개
- 지금까지의 unet3+ 코드 리팩토링하여 올립니다.

## 🔨 상세 작업 내용
> 상세 작업 내용 열거
- [x] train/valid split 되기 전 후 나눠서 리팩토링 진행
- [x] base : 
        1) base, deep super vision, CGM
        2) loss, optimizer
        3) soft, hard ensemble
        4) model_state_dict
- [x] split
        1) optuna
        2) loss, optimizer, image size
        3) 가중치 3가지
        4) augmentation
        5) encoder 변경
        6) k-fold
